common/thrift_client_pool.h

```diff
@@ -413,7 +413,7 @@ class ThriftClientPool {
     std::unique_ptr<T, Deleter> client(nullptr, deleter);
     auto ssl_ctx =
         ssl_ctx_ == nullptr ? nullptr : std::atomic_load_explicit(ssl_ctx_, std::memory_order_acquire);
-    event_loops_[idx].evb_->runImmediatelyOrRunInEventBaseThreadAndWait(
+    event_loops_[idx].evb_->runInEventBaseThreadAndWait(
```

Comment:
why rebase? I think this `runImmediatelyOrRunInEventBaseThreadAndWait` is the latest master, which introduce some flakiness to the unit tests.

Reply:
the reason why i changed to `runImmediatelyOrRunInEventBaseThreadAndWait` is that `runInEventBaseThreadAndWait` will introduce core dumps in u18 environment in some cases. 

Looking at the implementation of `runImmediatelyOrRunInEventBaseThreadAndWait`, it will run `runInEventBaseThreadAndWait ` when it is not called by the same evenbase or call function directly if it is in the same eventbase. so it is hard to understand why the tests failed

Reply:
@newpoo do you know why changing from `runInEventBaseThreadAndWait` to `runImmediatelyOrRunInEventBaseThreadAndWait ` will cause test failures? the underlying eventbase we pass to the asyncsocket should be the same but the tests were complaining not in the same eventbase?

Reply:
@richardzhangrui I don't have any clue off the top of may head. Have you looked at the test itself. Does it use the event base in any abnormal way?

Reply:
the tests look ok to me. let me do some more investigation